tokio-current-thread/src/lib.rs

```diff
@@ -702,12 +693,18 @@ impl Handle {
     /// This allows a caller to avoid creating the task if the call to `spawn`
     /// has a high likelihood of failing.
     pub fn status(&self) -> Result<(), SpawnError> {
-        if self.shut_down.get() {
+        if self.is_shut_down() {
             return Err(SpawnError::shutdown());
         }
 
         Ok(())
     }
+
+    fn is_shut_down(&self) -> bool {
+        // LSB of "num_futures" is the shutdown bit
+        let num_futures = self.num_futures.fetch_add(2, atomic::Ordering::SeqCst);
```

Comment:
I don't think you want to do a fetch_add here. Just a load should be fine.

Reply:
Ups, ofc. I pushed a fix.